### DISCUSSION TITLE
Help jar bundle/compile ugs-cli

### DISCUSSION THREAD

**Discussion by Milner99** [winder/Universal-G-Code-Sender#1867] [Help] [answered]
Hi,

The ugs-cli has run instructions to execute a shell java command against a .jar that isn't included in the ugs-cli folder. Please can someone help me bundle/compile that jar for use with Raspberry Pi?

thank you

**Comment by breiler** [comment]
The cli-variant of UGS is currently only included in the [classic edition](https://github.com/winder/Universal-G-Code-Sender/releases/download/v2.0.11/ugs-classic-2.0.11.zip), the script is called `start-cli.sh`

**Reply by Milner99** [reply]
Lovely, downloaded and moved to the Pi. Using the UI the carve is lovely. The CLI doesn't want to start? I fully expect it's me doing something wrong! - thoughts?
<img width="780" alt="Screenshot 2022-04-26 at 12 07 58" src="https://user-images.githubusercontent.com/1585406/165287124-6ca5fe73-66a8-4167-8cc7-b2b6270a0932.png">

**Reply by breiler** [reply]
Probably problems with file rights. Try to set it with the following:
```
chmod ugo+x start-cli.sh
```

As a side note, you shouldn't need to run the command with sudo. If you have problems accessing the USB-controller there might be some user rights missing to the pi-user. These can be added by adding the pi-user to the `dialout` group using the following command and a restart.
```
sudo adduser pi dialout
```

**Reply by Milner99** [reply]
Pi is a member of dial out already - the permission change helped and now I see this output. I'm looking to be able to launch a carve process using the .jar as per the help files (from my CMS using python or php exec()). 

Like this:
`java -jar ugs-cli-2.0-SNAPSHOT.jar --controller GRBL --port /dev/ttyUSB0 --baud 115200 --print-progressbar --file test.gcode`

<img width="531" alt="Screenshot 2022-04-26 at 12 22 02" src="https://user-images.githubusercontent.com/1585406/165289378-97850096-3e96-40d3-b663-2f8333c4086d.png">
 
Thank you for all your help.

**Comment by Milner99** [comment, marked as answer]
Figured it out, thank you for all your help @breiler 

The documentation says in the CLI folder 
`java -jar ugs-cli-2.0-SNAPSHOT.jar --controller GRBL --port /dev/ttyUSB0 --baud 115200 --print-progressbar --file test.gcode`
This however just launched UGS UI and works great but isn't CLI.

Can I suggest that the docs in the CLI folder (readme.md) get padded a little and/or changed to perhaps  say: Launch from command line example, use: 
`./start-cli.sh --controller GRBL --port /dev/ttyUSB0 --baud 115200 --print-progressbar --file test.gcode`

That is what I used to carve successfully from the terminal. 

Thanks again.
